### ISSUE TITLE
Deprecating Facebook Login support on Android WebViews

### ISSUE THREAD

**Issue by rafalzawadzki** [hungdev/react-native-instagram-login#99]
As [this document](https://developers.facebook.com/docs/facebook-login/android/deprecating-webviews) states,

> Beginning October 5, 2021, Facebook Login will no longer support using Android embedded browsers (WebViews) for logging in users.

Does this mean that Instagram login using this library will be affected?

**Comment by hungdev** [comment]
@rafalzawadzki no, Instagram login doesnt use webview from react native. It used WebView from react-native-webview

**Comment by rafalzawadzki** [comment]
@hungdev if I understand correctly, it does not matter where the WebView comes from - all [_embedded browsers_ are supposed to be blocked](https://developers.facebook.com/blog/post/2021/06/28/deprecating-support-fb-login-authentication-android-embedded-browsers/). And [`react-native-webview` also uses WebView internally](https://github.com/react-native-webview/react-native-webview/blob/master/android/src/main/java/com/reactnativecommunity/webview/WebViewConfig.java#L3).

This seems to be a general initiative because of security issues related to embedded browsers - eg. [here's a more elaborate explanation](https://developers.googleblog.com/2021/06/upcoming-security-changes-to-googles-oauth-2.0-authorization-endpoint.html) by Google who will disallow OAuth by the end of this month. Their ["browser policy"](https://developers.google.com/identity/protocols/oauth2/policies#browsers) states:

> A developer must not direct a Google OAuth 2.0 authorization request to an embedded user-agent under the developer's control. Embedded user-agents include, but are not limited to, software libraries that allow a developer to insert arbitrary scripts, alter the default routing of a request to the Google OAuth server, or access session cookies.

Nevertheless, it's not clear how exactly Facebook will recognize that a request comes from an embedded browser (headers etc), but from my understanding seems like this library might be affected. @hungdev could you elaborate on why exactly the WebView from `react-native-webview` should still work compared to WebView from RN?

**Comment by hungdev** [comment]
@rafalzawadzki  sorry, i misunderstood you. I thought where the problem came from webview of rn or react-native-webview.
And about the library will be affected?  I'm not sure if it will affected or not. 
The first one: Fb login doesnt support on android webview because they already has sdk fb login, and instagram doesnt. So i think we can still use the login api via webview, and fb still supports that.
Second one is we don't request to the Google OAuth server, we request api to fb, so i think it's not related to google.
In the end, I'm not sure about that.

**Comment by mohamedanwer123** [comment]
facebook prevent login from embedded browsers and we can't login instagram by facebook button
